app/controllers/dashboard_controller.rb

```diff
@@ -10,6 +10,45 @@ def index
     end
   end
 
+  def create
+    Dashboard::Manager.add_widget_to_user(User.current, params[:widget])
+    redirect_to root_path
+  end
+
+  def destroy
+    widget = Widget.find(params[:id])
```

Comment:
@tbrisker Using `find_resource` like this: https://github.com/theforeman/foreman/blob/develop/app/controllers/config_templates_controller.rb#L7 would delegate finding the appropriate permissions to the find_common concern.